libs/contrib/Network/Socket.idr

```diff
@@ -186,199 +234,380 @@ saString Nothing = ""
+sendBuf : (sock : Socket)
+       -> (ptr  : BufPtr)
+       -> (len  : ByteLength)
+       -> IO (Either SocketError ResultCode)
 sendBuf sock (BPtr ptr) len = do
-  send_res <- foreign FFI_C "idrnet_send_buf" (Int -> Ptr -> Int -> IO Int) (descriptor sock) ptr len
-  if send_res == (-1) then
-    map Left getErrno
-  else
-    return $ Right send_res
+  send_res <- foreign FFI_C "idrnet_send_buf"
+                      (Int -> Ptr -> Int -> IO Int)
+                      (descriptor sock) ptr len
 
-recvBuf : Socket -> BufPtr -> ByteLength -> IO (Either SocketError ByteLength)
+  if send_res == (-1)
+   then map Left getErrno
+   else return $ Right send_res
+
+||| Receive data from a given memory location.
+|||
+||| Returns on failure a `SocketError`
+||| Returns on success the `ResultCode`
+|||
+||| @sock The socket on which to receive the message.
+||| @ptr  The location containing the data to receive.
+||| @len  How much of the data to receive.
+recvBuf : (sock : Socket)
+       -> (ptr  : BufPtr)
+       -> (len  : ByteLength)
+       -> IO (Either SocketError ResultCode)
 recvBuf sock (BPtr ptr) len = do
-  recv_res <- foreign FFI_C "idrnet_recv_buf" (Int -> Ptr -> Int -> IO Int) (descriptor sock) ptr len
-  if (recv_res == (-1)) then
-    map Left getErrno
-  else
-    return $ Right recv_res
+  recv_res <- foreign FFI_C "idrnet_recv_buf"
+                      (Int -> Ptr -> Int -> IO Int)
+                      (descriptor sock) ptr len
+
+  if (recv_res == (-1))
+    then map Left getErrno
+    else return $ Right recv_res
 
+||| Send a message.
+|||
+||| Returns on failure a `SocketError`
+||| Returns on success the `ResultCode`
+|||
+||| @sock The socket on which to send the message.
+||| @addr Address of the recipient.
+||| @port The port on which to send the message.
+||| @msg  The message to send.
 export
-sendTo : Socket -> SocketAddress -> Port -> String -> IO (Either SocketError ByteLength)
+sendTo : (sock : Socket)
+      -> (addr : SocketAddress)
+      -> (port : Port)
+      -> (msg  : String)
+      -> IO (Either SocketError ByteLength)
 sendTo sock addr p dat = do
   sendto_res <- foreign FFI_C "idrnet_sendto"
                    (Int -> String -> String -> Int -> Int -> IO Int)
                    (descriptor sock) dat (show addr) p (toCode $ family sock)
-  if sendto_res == (-1) then
-    map Left getErrno
-  else
-    return $ Right sendto_res
 
-sendToBuf : Socket -> SocketAddress -> Port -> BufPtr -> ByteLength -> IO (Either SocketError ByteLength)
+  if sendto_res == (-1)
+    then map Left getErrno
+    else return $ Right sendto_res
+
+||| Send a message stored in some buffer.
+|||
+||| Returns on failure a `SocketError`
+||| Returns on success the `ResultCode`
+|||
+||| @sock The socket on which to send the message.
+||| @addr Address of the recipient.
+||| @port The port on which to send the message.
+||| @ptr  A Pointer to the buffer containing the message.
+||| @len  The size of the message.
+export
```

Comment:
`sendToBuf` and `recvFromBuf` were previously not exported.

It would make sense to either remove those `export` or add some to `sendBuf` and `recvBuf`.